client/blocks/sharing-preview-pane/index.jsx

```diff
@@ -48,6 +58,11 @@ class SharingPreviewPane extends PureComponent {
 	constructor( props ) {
 		super( props );
 
+		if ( ! props.isMastodonEligible ) {
+			const { mastodon, ...rest } = props.services;
+			props.services = rest;
```

Comment:
We should never mutate the props. It can result in unexpected behavior/errors. 

<img width="780" alt="Screenshot 2023-06-02 at 12 43 01 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/2f8f5077-0b4a-4d9c-ae2b-b064498155fd">

CC: @kraftbj

Reply:
There are two issues here:
1. Props are being mutated.
2. propTypes mention `services` as an array and here we are assign an object to it. `props.services` is being used as an array in the component.

Reply:
Here is the PR to fix the issue #77737